src/libraries/Microsoft.PowerFx.Core/Lexer/Tokens/IdentToken.cs

```diff
@@ -21,7 +21,7 @@ public IdentToken(string val, Span span)
             : this(val, span, false, false)
         {
             Contracts.AssertValue(val);
-            Contracts.Assert(val.Length == span.Lim - span.Min);
+            Contracts.Assert(val.Length == span.Lim - span.Min || val == "Concatenate");
```

Comment:
>|| val == "Concatenate" [](http://example.com/codeflow?start=63&length=23)

This deserves a comment!

Reply:
May deserve a const field so this value is shared with ParseStringInterpolation(). That also helps associate the code snippets together to self-document.

Reply:
Added a comment.